src/runtime_src/core/pcie/linux/device_linux.cpp

```diff
@@ -888,6 +891,72 @@ struct accel_deadlock_status
   }
 };
 
+// structure to get Aie tiles status version info
+// This version is used for handshake b/w userspace and driver
+struct aie_status_version
+{
+  using result_type = query::aie_status_version::result_type;
+
+  static result_type
+  get(const xrt_core::device* device, key_type key)
+  {
+    result_type version{0};
+
+    xocl::shim *drv = xocl::shim::handleCheck(device->get_user_handle());
+    if(!drv)
+      throw std::runtime_error("Unable to get shim handle of the device");
+      
+    drv->get_aie_status_version_info(version.major, version.minor);
+
+    return version;
+  }
+};
+
+// Structure to get device specific Aie tiles information like
+// Total rows, cols and num of core, mem, shim rows and thier start row num
+// num of dma channels, locks, events etc
+struct aie_tiles_stats
+{
+  using result_type = query::aie_tiles_stats::result_type;
+
+  static result_type
+  get(const xrt_core::device* device, key_type key)
+  {
+    uint32_t size = sizeof(result_type);
+    std::vector<char> buf(size, 0);
+
+    xocl::shim *drv = xocl::shim::handleCheck(device->get_user_handle());
+    if(!drv)
+      throw std::runtime_error("Unable to get shim handle of the device");
+      
+    drv->get_aie_tiles_info(buf.data(), size);
+
+    return *(reinterpret_cast<result_type*>(buf.data()));
+  }
+};
+
+// structure to get aie tiles status raw buffer
+struct aie_cols_status_info
+{
+  using result_type = boost::any;
+
+  static result_type
+  get(const xrt_core::device* device, key_type key, const boost::any& param)
+  {
+    auto data = boost::any_cast<xrt_core::query::aie_cols_status_info::meta_data>(param);
+
+    std::vector<char> buf(data.col_size * data.num_cols, 0);
+
+    xocl::shim *drv = xocl::shim::handleCheck(device->get_user_handle());
+    if(!drv)
+      throw std::runtime_error("Unable to get shim handle of the device");
+      
+    drv->get_aie_col_info(buf.data(), data.col_size * data.num_cols, data.start_col, data.num_cols);
+
+    return buf;
+  }
+};
```

Comment:
Since they want ioctl due to security reasons, lets make the Reports and the sysfs nodes available only for the mgmtpf. Then you need sudo to get any info

Reply:
At present its a TODO on getting the data right, we will have discussion on how data can be provided. I will remove shim calls for now.